x/gamm/twap/hook_listener.go

```diff
@@ -1,14 +1,22 @@
 package twap
 
 import (
-	"fmt"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	epochtypes "github.com/osmosis-labs/osmosis/v10/x/epochs/types"
 	"github.com/osmosis-labs/osmosis/v10/x/gamm/types"
 )
 
+func (k Keeper) MigrateExistingPools(ctx sdk.Context, poolIds []uint64) error {
```

Comment:
I wanted this file to be `listeners` for things that could trigger a state migration. Maybe I just change the file name to listeners.go?

I'm imagining we try to to change the entire Hook system into a publisher subscriber system

Reply:
> Maybe I just change the file name to listeners.go?

I like `listeners.go` since hook and listener feel to be synonymous. Also, I think `listener` is frequently used when speaking of pub-sub systems.

> I wanted this file to be listeners for things that could trigger a state change

But this specific function is a one-time migration that is done during the upgrade. Why would we ever use pub-sub for this logic?

Also, I was referring to existing examples such as:
https://github.com/osmosis-labs/osmosis/blob/3b9e72c7e243eabb9c2bd5c91c1f3ced044cb3dd/x/lockup/keeper/migration.go#L40

where upgrade migrations logic exists in the keeper's `migrations.go` file

Reply:
hrmm yeah your right the migraiton being here is weird.

I just realized that we can do this in InitGenesis, which is likely the correct entry point. WIll move it there.

Reply:
ah rip we can't do it in InitGenesis without adding to the gamm keeper interface, which seems worse.

I feel like keeping it in listeners.go, as what an external module can trigger logic for, is ok personally. I'd prefer not blocking on where this function lives, as it will ultimately be deleted later.

Perhaps we can call it InitGenesisAtMigration?

Reply:
We agreed to leaves as is offline - I guess we can discuss this in a separate issue later